contracts/modules/assessment/Assessment.sol

```diff
@@ -2,515 +2,365 @@
+  /// @notice Allows for the early closing of a claim's voting period.
+  /// @dev Can only be called if all assigned assessors have cast their votes.
+  ///      Must be called by an active assessor for the claim to prevent potential front-running.
+  ///      Sets the assessment's `votingEnd` to the current block timestamp.
+  /// @param claimId The unique identifier for the claim.
+  function closeVotingEarly(uint claimId) override external {
+    // an assessor on the claim must call this to prevent front-running
+    (, Assessment memory assessment) = _validateAssessor(claimId, msg.sender);
+    require(block.timestamp < assessment.votingEnd, VotingAlreadyClosed());
+
+    uint[] memory assessors = getGroupAssessors(assessment.assessingGroupId);
+    uint groupSize = assessors.length;
+    uint totalVotesFromGroup = 0;
+
+    for (uint i = 0; i < groupSize; i++) {
+      if (_ballots[assessors[i]][claimId].timestamp > 0) {
+        totalVotesFromGroup++;
+      } else {
+        // can break early because we need everybody to vote
+        break;
       }
-
-      emit FraudProcessed(vote.assessmentId, assessor, poll);
-      assessments[vote.assessmentId].poll = poll;
     }
 
-    // Burns an assessor only once for each merkle tree root, no matter how many times this function
-    // runs on the same account. When a transaction is too big to fit in one block, it is batched
-    // in multiple transactions according to voteBatchSize. After burning the tokens, fraudCount
-    // is incremented. If another merkle root is submitted that contains the same addres, the leaf
-    // should use the updated fraudCount stored in the Stake struct as input.
-    if (fraudCount == _stake.fraudCount) {
-      // Make sure this doesn't revert if the stake amount is already subtracted due to a previous
-      // burn from a different merkle tree.
-      burnAmount = burnAmount > _stake.amount ? _stake.amount : burnAmount;
-      _stake.amount -= burnAmount;
-      _stake.fraudCount++;
-
-      // TODO: consider burning the tokens in the token controller contract
-      _ramm().updateTwap();
-      nxm.burn(burnAmount);
-    }
+    require(totalVotesFromGroup == groupSize, NotEverybodyVoted());
 
-    if (uint104(processUntil) > _stake.rewardsWithdrawableFromIndex) {
-      _stake.rewardsWithdrawableFromIndex = uint104(processUntil);
-    }
+    assessment.votingEnd = block.timestamp.toUint32();
+    _assessments[claimId] = assessment;
 
-    stakeOf[assessor] = _stake;
+    emit AssessmentVotingEndChanged(claimId, assessment.votingEnd);
   }
 
-  /* ========== DEPENDENCIES ========== */
+  /// @notice Resets the voting period for a claim, starting a new full voting window.
+  /// @dev Can only be called by the Governor contract. Reverts if the assessment's cooldown period has already passed.
+  /// @param claimId The unique identifier for the claim.
+  function resetVotingPeriod(uint claimId) override external onlyContracts(C_GOVERNOR) {
```

Comment:
naming: consider `extend` instead of `reset`. Resseting sounds like setting it to a default, while extending seems more appropriate here.